source/common/common/matchers.cc

```diff
@@ -130,13 +130,30 @@ PathMatcherConstSharedPtr PathMatcher::createExact(const std::string& exact, boo
   return std::make_shared<const PathMatcher>(matcher);
 }
 
-PathMatcherConstSharedPtr PathMatcher::createPrefix(const std::string& prefix, bool ignore_case) {
+namespace {
+PathMatcherConstSharedPtr createPrefixPathMatcher(const std::string& prefix, bool ignore_case) {
   envoy::type::matcher::v3::StringMatcher matcher;
   matcher.set_prefix(prefix);
   matcher.set_ignore_case(ignore_case);
   return std::make_shared<const PathMatcher>(matcher);
 }
 
+} // namespace
+
+PathMatcherConstSharedPtr PathMatcher::createPrefix(const std::string& prefix, bool ignore_case) {
+  if (prefix.empty()) {
+    static const PathMatcherConstSharedPtr emptyPrefixPathMatcher =
+        createPrefixPathMatcher("", false);
+    return emptyPrefixPathMatcher;
```

Comment:
I guess that to use `CONSTRUCT_ON_FIRST_USE`, one will need to pass to it a function `createEmptyPrefixPathMatcher()` that essentially does `return createPrefixPathMatcher("", false);`.

I guess it complicates the readability of this code.